pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -60,6 +60,7 @@ const (
 	servicePortStatPattern     = "%SERVICE_PORT%"
 	servicePortNameStatPattern = "%SERVICE_PORT_NAME%"
 	subsetNameStatPattern      = "%SUBSET_NAME%"
+	k8sDomain                  = ".svc.cluster.local"
```

Comment:
and herein lies the problem.. k8s allows customizing the domain name!!

Reply:
Thats why we added the proxy.dnsdomain. May be the easier fix would be for the proxy to tell us whether it is on k8s or not. This ofcourse can be easily obatained from the service instances of the proxy [to the service to the service attribute which contains the platform where that service is present]